package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -0,0 +1,777 @@
+              <donor index (0-based)>,
+              <acceptor index (0-based)>, <donor string>, <acceptor string>,
+              <distance>, <angle>
+           ],
+           ....
+        ],
+        [ # frame 2
+          [ ... ], [ ... ], ...
+        ],
+        ...
+    ]
+
+Using the :meth:`WaterBridgeAnalysis.generate_table` method one can reformat
+the results as a flat "normalised" table that is easier to import into a
+database or dataframe for further processing.
+:meth:`WaterBridgeAnalysis.save_table` saves the table to a pickled file. The
+table itself is a :class:`numpy.recarray`.
+
+Detection of water bridges
+---------------------------
+Water bridges are recorded if a bridging water simultaneously form two hydrogen bonds with selection 1 and selection 2.
+
+Hydrogen bonds are detected as is described in \ :class:`~MDAnalysis.analysis.hbonds.hbond_analysis.HydrogenBondAnalysis`, see \ :ref:`Detection-of-hydrogen-bonds`.
+
+The lists of donor and acceptor names can be extended by providing lists of
+atom names in the `donors` and `acceptors` keywords to
+:class:`WaterBridgeAnalysis`. If the lists are entirely inappropriate
+(e.g. when analysing simulations done with a force field that uses very
+different atom names) then one should either use the value "other" for `forcefield`
+to set no default values, or derive a new class and set the default list oneself::
+
+ class WaterBridgeAnalysis_OtherFF(WaterBridgeAnalysis):
+       DEFAULT_DONORS = {"OtherFF": tuple(set([...]))}
+       DEFAULT_ACCEPTORS = {"OtherFF": tuple(set([...]))}
+
+Then simply use the new class instead of the parent class and call it with
+`forcefield` = "OtherFF". Please also consider to contribute the list of heavy
+atom names to MDAnalysis.
+
+How to perform WaterBridgeAnalysis
+-----------------------------------
+
+All water bridges between arginine and aspartic acid can be analysed with ::
+
+  import MDAnalysis
+  import MDAnalysis.analysis.hbonds
+
+  u = MDAnalysis.Universe('topology', 'trajectory')
+  w = MDAnalysis.analysis.hbonds.WaterBridgeAnalysis(u, 'resname ARG', 'resname ASP')
+  w.run()
+
+The results are stored as the attribute
+:attr:`WaterBridgeAnalysis.timeseries`; see :ref:`Analysis Output` for the
+format.
+
+An example of using the :attr:`~WaterBridgeAnalysis.timeseries` would be detecting the precentage of time a certain water bridge exits. ::
+
+  w.timeseries
+  #[ # frame 1
+  #  # A water bridge SOL2 links O from ARG1 and ASP3
+  # [[0,1,'ARG1O',  'SOL2HW1',3.0,180],
+  #  [2,3,'SOL2HW2','ASP3O',  3.0,180],
+  # ],
+  #  # frame 2
+  #  # No water bridge detected
+  # []
+  #]
+
+  water_bridge_presence = []
+  for frame in w.timeseries:
+      if frame:
+          water_bridge_presence.append(True)
+      else:
+          water_bridge_presence.append(False)
+  print(sum(water_bridge_presence)/len(water_bridge_presence))
+  # 0.5
+  # Water bridge can be detected in 50% of the time
+
```

Comment:
Have you got an example where you process a large number of bridging waters, e.g., when the two selections are two different proteins and you are looking at all bridging waters in the interface because you want to analyze water-mediated protein-protein interactions? For instance, I'd like to know what  the most important residue pairs (protein_1.residue_i - water - protein_2.residue_j) are.

This is the kind of example that I am really looking for here because using your data structure for those kinds of questions is much more challenging than using it for a simple analysis of two well-defined residues.

Reply:
I am happy with an example that uses a pandas dataframe! (This would also show off the usefulness of `generate_table()`.

Reply:
@orbeckst Sorry, I know that it might be a bit challenging to reconstruct the network from the timeseries. locally, I have written dozens of ugly functions that do all sorts of useful and useless things.
However, I do have a function which can answer your question. I have implemented it to replace the HydrogenBondAnalysis.count_by_type() method and the supplementary the doc.